### PR TITLE
Convert unit test to a NodeJS project.

### PR DESCRIPTION
**Pull request by martinbednar**

### CODE REVIEW

File: tests/unit_tests/package.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  "name": "unit_tests",
+  "version": "1.0.0",
+  "description": "JSR unit tests.",
+  "main": "",
+  "scripts": {
+    "test": "./start_unit_tests.sh",
+    "start": "./start_unit_tests.sh"
+  },
+  "author": "Martin Bednar, ibednar@fit.vutbr.cz",
+  "license": "GNU General Public License",
```
Comment (polcak) [inline tests/unit_tests/package.json:11]: The license is missing the version. I suggest "3 or later" to be compatible with the rest of the project

File: tests/unit_tests/config/jasmine.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{ "spec_dir": "tmp", "spec_files": [ <<SPEC_FILES>> ], "helpers": [ ], "stopSpecOnExpectationFailure": false, "random": true }
```
Comment (polcak) [inline tests/unit_tests/config/jasmine.json:1]: Do you mean CR or LF?

File: tests/unit_tests/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Run unit tests
+
+## on Linux
+
+1. Install NodeJS ([https://nodejs.org]()).
+1. Install `jq` and another necessary tools (e.g. `sed`).
+1. Open Terminal.
+1. Run `npm install`.
+1. Run `npm test`.
+
```
Comment (polcak) [inline tests/unit_tests/README.md:10]: For compatibility with doxygen, we might need using 1., 2., 3., ...

File: tests/unit_tests/README.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+## on Windows
+
+1. Install Windows Subsystem for Linux (WSL): https://docs.microsoft.com/en-us/windows/wsl/install-win10.
+1. Convert EOL in the script *./start_unit_tests.sh* from Windows (CR LF) to Unix (LF) - you can use the tool `dos2unix` in WSL to convert CR LF to LF.
+1. Follow the instructions for Linux. Install NodeJS and run the following commands in WSL.
```
Comment (polcak) [inline tests/unit_tests/README.md:15]: See the doxygen comment above

**Review by polcak — CHANGES_REQUESTED** [review]
It seems to me that this should be merged to both geo and master branch?

File: tests/unit_tests/package.json (hunk 1 above)
Comment (martinbednar) [reply tests/unit_tests/package.json]: Thanks for your advice. I was not sure about the license. I have resolve this suggestion in the commit #[2bf11cd2770572099735767379db471253c59a66](https://github.com/polcak/jsrestrictor/pull/84/commits/2bf11cd2770572099735767379db471253c59a66).

File: tests/unit_tests/config/jasmine.json (hunk 2 above)
Comment (martinbednar) [reply tests/unit_tests/config/jasmine.json]: I am sorry, I have set wrong EOL. Corrected in the commit [dc45a44ae1b40c8058f73d083d4fc35f107d9004](https://github.com/polcak/jsrestrictor/pull/84/commits/dc45a44ae1b40c8058f73d083d4fc35f107d9004)

File: tests/unit_tests/README.md (hunk 3 above)
Comment (martinbednar) [reply tests/unit_tests/README.md:10]: Fixed in the commit [946a9a0b32ed95e838f5820b47c188f75b086223](https://github.com/polcak/jsrestrictor/pull/84/commits/946a9a0b32ed95e838f5820b47c188f75b086223).

File: tests/unit_tests/README.md (hunk 4 above)
Comment (martinbednar) [reply tests/unit_tests/README.md]: Fixed in the commit [946a9a0b32ed95e838f5820b47c188f75b086223](https://github.com/polcak/jsrestrictor/pull/84/commits/946a9a0b32ed95e838f5820b47c188f75b086223).

**Comment by martinbednar** [comment]
> It seems to me that this should be merged to both geo and master branch?

Yes, I agree with you. Changes in the pull request #84 should be merged to both geo and master branch. Should I create second pull request to the master branch or you can merge it yourself?

**Pull request merged**